src/vs/workbench/contrib/chat/common/chatServiceImpl.ts

```diff
@@ -395,6 +395,12 @@ export class ChatService extends Disposable implements IChatService {
 		const model = this.instantiationService.createInstance(ChatModel, someSessionHistory, location);
 		this._sessionModels.set(model.sessionId, model);
 		this.initializeSession(model, token);
+
+		if (someSessionHistory) {
+			if ('isToolsAgentModeEnabled' in someSessionHistory) {
```

Comment:
This is a little funky because it will run for all sessions, not just transferred ones, right? Since this is persisted somewhere else too, that could be confusing. Maybe we can just save this flag on the `IChatTransferredSessionData` and only set it when loading the transferred session?